crates/node-core/src/metrics/prometheus_exporter.rs

```diff
@@ -70,6 +70,12 @@ async fn start_endpoint<F: Hook + 'static>(
     let server =
         Server::try_bind(&listen_addr).wrap_err("Could not bind to address")?.serve(make_svc);
 
+    let server = server.with_graceful_shutdown(async {
+        tokio::signal::ctrl_c()
+            .await
+            .wrap_err("Failed to listen for shutdown signal")
+    });
```

Comment:
The server should be spawned with https://github.com/paradigmxyz/reth/blob/528f1e904766bcafc3aa1e109fda27cfd991b5be/crates/tasks/src/lib.rs#L538C12-L538C47 and the `GracefulShutdown` signal the task is given should be used

we should not use tokio::signal::ctrl_c here since there are other reasons the node might shut down

Reply:
> The server should be spawned with https://github.com/paradigmxyz/reth/blob/528f1e904766bcafc3aa1e109fda27cfd991b5be/crates/tasks/src/lib.rs#L538C12-L538C47 and the `GracefulShutdown` signal the task is given should be used
> 
> we should not use tokio::signal::ctrl_c here since there are other reasons the node might shut down

working on it